torch/utils/cpp_extension.py

```diff
@@ -575,10 +575,12 @@ def __init__(self, *args, **kwargs):
         return cls_with_options
 
     def __init__(self, *args, **kwargs) -> None:
-        super().__init__(*args, **kwargs)
         self.no_python_abi_suffix = kwargs.get("no_python_abi_suffix", False)
-
         self.use_ninja = kwargs.get('use_ninja', True)
+
+        filtered_kwargs = {kw: val for kw, val in kwargs.items() if kw not in ["no_python_abi_suffix", "use_ninja"]}
```

Comment:
I wonder if better way to address that is to define user_options (or boolean_options)
Not sure how long something like that were supported though, see https://github.com/pypa/setuptools/blob/7c859e017368360ba66c8cc591279d8964c031bc/setuptools/_distutils/command/build_ext.py#L100